akka-http-core/src/main/scala/akka/http/impl/settings/ParserSettingsImpl.scala

```diff
@@ -55,6 +56,12 @@ private[akka] final case class ParserSettingsImpl(
   override def headerValueCacheLimit(headerName: String): Int =
     headerValueCacheLimits.getOrElse(headerName, defaultHeaderValueCacheLimit)
 
+  override def maxContentLength: Long =
+    maxContentLengthSetting.getOrElse(
```

Comment:
This is quite unfortunate but a consequence of keeping maxContentLength in ParserSettings. I'll have another quick try of just moving the value out of there and directly into `ServerSettings` / `ClientConnectionSettings`.

Reply:
I tried to push the setting to ServerSettings / ClientConnectionSettings but that really doesn't give us a lot: people still have to use `ParserSettings.forServer` / `ParserSettings.forClient` to get the right set of settings.

I guess that's the way it is. If you programmatically use e.g. `ServerSettings(system)` and then modify those settings, you will have kind of a complex stack of settings.

E.g. if we merge the per-host settings you will have this stack:

host-specific programmatic parsing settings -> host-specific application.conf parsing settings -> host-connection-pool parsing settings -> client parsing settings -> general parsing settings

which is already quite complex.

Maybe we need to provide the old setting here hard-coded ? At least for a deprecation period?

Reply:
I had not addressed this point.

IIRC, it means that code patterns where users used `ClientConnectionSettings(system).withParsingSettings(ParsingSettings(system).withXYZ())` will now throw a runtime exception when used.

Reply:
Created #3179 so we don't forget about this.